martin/src/cog/source.rs

```diff
@@ -456,6 +480,79 @@ fn get_origin(
     }
 }
 
+fn get_full_resolution(
+    pixel_scale: Option<&[f64]>,
+    transformation: Option<&[f64]>,
+    path: &Path,
+) -> Result<[f64; 2], CogError> {
+    match (pixel_scale, transformation) {
+        // ModelPixelScaleTag = (ScaleX, ScaleY, ScaleZ)
+        (Some(scale), _) => Ok([scale[0], scale[1]]),
+        (_, Some(matrix)) => {
+            let mut x_res =
+                (matrix[0] * matrix[0] + matrix[4] * matrix[4] + matrix[8] * matrix[8]).sqrt();
+            x_res = x_res.copysign(matrix[0]);
+            let mut y_res =
+                (matrix[1] * matrix[1] + matrix[5] * matrix[5] + matrix[9] * matrix[9]).sqrt();
+            // A positive y_res indicates that model space Y cordinates decrease as raster space J indices increase. This is the standard vertical relationship between raster space and model space
+            y_res = y_res.copysign(-matrix[5]);
+            Ok([x_res, y_res]) // drop the z scale directly as we don't use it
+        }
+        (None, None) => Err(CogError::GetFullResolutionFailed(path.to_path_buf())),
+    }
+}
+
+fn raster2model(i: u32, j: u32, matrix: &[f64]) -> (f64, f64) {
+    let i = f64::from(i);
+    let j = f64::from(j);
+    let x = matrix[3] + (matrix[0] * i) + (matrix[1] * j);
+    let y = matrix[7] + (matrix[4] * i) + (matrix[5] * j);
+    (x, y)
+}
+
+fn get_extent(
```

Comment:
```suggestion
/// Computes an `[min_x, min_y, max_x, max_y]` tuple and returns the geographic extent (bounding box) of the given spatial layer. 
///
/// Applies a transformation matrix to corner pixels if provided;
/// otherwise, computes extent from origin and raster size in model units.
///
/// # Parameters
/// - `origin`: Top-left origin `[x, y, z]` in model space.
/// - `transformation`: Optional pixel-to-model transformation matrix.
/// - `full_width_pixel`, `full_height_pixel`: Raster dimensions in pixels.
/// - `full_width`, `full_height`: Raster size in model units.
fn get_extent(
```

Reply:
@sharkAndshark
Except adding a doc comment for this method (and maybe the other two also), I am fine with this.
Does not have to be such a long doc comment

```suggestion
/// Computes an `[min_x, min_y, max_x, max_y]` tuple and returns the geographic extent (bounding box) of the given spatial layer. 
///
/// Applies a transformation matrix to corner pixels if provided;
/// otherwise, computes extent from origin and raster size in model units.
fn get_extent(
```